equistore-torch/src/register.cpp

```diff
@@ -180,15 +181,38 @@ TORCH_LIBRARY(equistore, m) {
         .def("print", &TensorMapHolder::print, DOCSTRING,
             {torch::arg("max_keys")}
         )
-        // TODO
-        // .def_pickle(
-        //     // __getstate__
-        //     [](const torch::intrusive_ptr<TorchTensorMap>& self) -> std::string {
-        //          // TODO
-        //     },
-        //     // __setstate__
-        //     [](std::string state) -> torch::intrusive_ptr<TorchCalculator> {
-        //         // TODO
-        //     })
+        .def_pickle(
+            // __getstate__
+            [](const TorchTensorMap& self) -> torch::Tensor {
+                auto buffer = equistore::TensorMap::save_buffer(self->as_equistore());
+                // move the buffer to the heap so it can escape this function
```

Comment:
can we explain this a bit more? It was nontrivial for me that from_blob does not take ownership, that is why we need to put on the heap.

```suggestion
                // move the buffer to the heap so it can escape this function
                // since from_blob will not take ownership over the data
```

Reply:
done!